pkg/users/group-create-dialog.js

```diff
@@ -0,0 +1,152 @@
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from 'cockpit';
+import React from 'react';
+
+import { Form, FormGroup, TextInput } from '@patternfly/react-core';
+import { show_modal_dialog, apply_modal_dialog } from "cockpit-components-dialog.jsx";
+
+import { has_errors, is_valid_char_name } from "./dialog-utils.js";
+
+const _ = cockpit.gettext;
+
+function GroupCreateBody({ state, errors, change }) {
+    const {
+        name, id,
+    } = state;
+
+    return (
+        <Form isHorizontal onSubmit={apply_modal_dialog}>
+            <FormGroup label={_("Name")}
+                       helperTextInvalid={errors && errors.name}
+                       validated={(errors && errors.name) ? "error" : "default"}
+                       fieldId="groups-create-name">
+                <TextInput id="groups-create-name"
+                           validated={(errors && errors.name) ? "error" : "default"}
+                           value={name} onChange={value => change("name", value)} />
+            </FormGroup>
+
+            <FormGroup label={_("ID")}
+                       hasNoPaddingTop
+                       helperTextInvalid={errors && errors.id}
+                       validated={(errors && errors.id) ? "error" : "default"}
+                       isStack
+                       fieldId="groups-create-id">
+                <TextInput id="groups-create-id"
+                           validated={(errors && errors.id) ? "error" : "default"}
+                           value={id} onChange={value => change("id", value)} />
+            </FormGroup>
+        </Form>
+    );
+}
+
+function validate_name(name, groups) {
+    if (!name)
+        return _("No group name specified");
+
+    for (let i = 0; i < name.length; i++) {
+        if (!is_valid_char_name(name[i]))
+            return _("The group name can only consist of letters from a-z, digits, dots, dashes and underscores.");
+    }
+
+    for (let k = 0; k < groups.length; k++) {
+        if (groups[k].name == name)
+            return _("A group with this name already exists");
+    }
+
+    return null;
+}
+
+export function group_create_dialog(groups, setGroupsCardExpanded) {
+    cockpit.file("/etc/login.defs").read()
+            .then(logindef => {
+                const gids = groups
+                        .filter(g => g.name !== 'nobody')
+                        .map(group => group.gid);
+                const min_gid = parseInt(logindef.match(/^GID_MIN\s+(\d+)/m)[1] || 100);
+                const max_gid = parseInt(logindef.match(/^GID_MAX\s+(\d+)/m)[1] || 65534);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18136-20230119-140210-b1c4d1e7-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)